examples/custom_layout_manager/custom_layout.rs

```diff
@@ -0,0 +1,186 @@
+use crate::TOTAL_COLORS;
+use gtk::glib;
+use gtk::prelude::*;
+use gtk::subclass::prelude::*;
+
+const N_GRID_COLUMNS: u32 = 4;
+
+mod imp {
+    use super::*;
+    use std::cell::{Cell, RefCell};
+    #[derive(Debug)]
+    pub struct CustomLayout {
+        pub position: Cell<f64>,
+        pub child_pos: RefCell<Vec<i32>>,
+    }
+    impl Default for CustomLayout {
+        fn default() -> Self {
+            let mut positions = Vec::new();
+            for i in 0..TOTAL_COLORS {
+                positions.push(i);
+            }
+            Self {
+                position: Cell::new(0f64),
+                child_pos: RefCell::new(positions),
+            }
+        }
+    }
+
+    #[glib::object_subclass]
+    impl ObjectSubclass for CustomLayout {
+        const NAME: &'static str = "CustomLayout";
+        type Type = super::CustomLayout;
+        type ParentType = gtk::LayoutManager;
+    }
+
+    impl ObjectImpl for CustomLayout {}
+    impl LayoutManagerImpl for CustomLayout {
+        fn request_mode(
+            &self,
+            _layout_manager: &Self::Type,
+            _widget: &gtk::Widget,
+        ) -> gtk::SizeRequestMode {
+            gtk::SizeRequestMode::ConstantSize
+        }
+
+        fn measure(
+            &self,
+            _layout_manager: &Self::Type,
+            widget: &gtk::Widget,
+            orientation: gtk::Orientation,
+            _for_size: i32,
+        ) -> (i32, i32, i32, i32) {
+            let mut min_size = 0;
+            let mut nat_size = 0;
+
+            let mut child = widget.first_child().unwrap();
+            loop {
+                if !child.should_layout() {
+                    continue;
+                }
+
+                let (child_min, child_nat, _, _) = child.measure(orientation, -1);
+                min_size = min_size.max(child_min);
+                nat_size = nat_size.max(child_nat);
+                if child.next_sibling().is_none() {
```

Comment:
```rust
if let Some(next_child) = child.next_sibling() {
    child = next_child;
} else {
    break;
}
```

Reply:
Oups, those are reasons to not write code that late at night :)